### PR TITLE
Bug 1798482: Add CRD generator documentation to Readme

### LINKED ISSUE #572: Missing documentation for API schema policy
In slack today I was alerted to the possibility that an OpenShift component to which I contribute may not be in compliance with a policy requiring a schema for its APIs. I don't know the details of that policy, how to tell if it actually affects the component I'm working on, or what sort of code (or other) changes might be needed in order to comply. There does not appear to be any documentation about that here in this repo, so I'm filing this ticket to ask someone to write all of that down so I can ensure someone takes care of any gaps we have.

I don't think we link to slack conversations in GitHub issues, so ping me and I'll share a link privately if you would like more details.

### PR DESCRIPTION
**Pull request by damemi**
Adds information, links, and steps to set up and use the crd schema generator

Fixes #572 
/cc @dhellmann

### CODE REVIEW

**Comment by damemi** [comment]
cc @smarterclayton @sttts could you please review?

File: README.md [hunk 1]
```diff
@@ -29,3 +29,41 @@ Then do the following with your openshift/origin pull request:
 1. Drop the TMP commit (pointing glide back at openshift/api#master)
 2. Rerun `hack/update-deps.sh` and update your `bump(*)` commit
 3. It can then be tagged and merged by CI
+
+## generating CRD schemas
+
+Since Kubernetes 1.16, every CRD created in `apiextensions.k8s.io/v1` is required to have a [structural OpenAPIV3 schema](https://kubernetes.io/blog/2019/06/20/crd-structural-schema/). The schemas provide server-side validation for fields, as well as providing the descriptions for `oc explain`. Structural requirements build on that to provide enhanced security in objects' persistence in etcd. As we host many of our CRDs in this repo along with their corresponding Go types we also require them to have schemas. However, the following instructions apply for CRDs that are not hosted here as well.
```
Comment (sttts) [inline README.md]: > Structural requirements build on that to provide enhanced security in objects' persistence in etcd

Not sure I can follow. You mean: "Moreover, schemas ensure structural consistency of data in etcd. Without anything can be stored in a resource which can have security implications."

File: README.md [hunk 2]
```diff
@@ -29,3 +29,41 @@ Then do the following with your openshift/origin pull request:
...
+
+These schemas are often very long and complex, and should not be written by hand. For OpenShift, we provide Makefile targets in [library-go's alpha-build-machinery](https://github.com/openshift/library-go/tree/master/alpha-build-machinery) which generate the schema.
+
+If you make a change to a CRD type in this repo, simply calling `make update-codegen-crds` should regenerate all CRDs. If yours is not updated, ensure that the path to its API is included in our [calls to the Makefile targets](https://github.com/openshift/api/blob/release-4.5/Makefile#L17-L29).
```
Comment (sttts) [inline README.md]: regenerate all CRD schemas and update the manifests.

File: README.md [hunk 3]
```diff
@@ -29,3 +29,41 @@ Then do the following with your openshift/origin pull request:
 1. Drop the TMP commit (pointing glide back at openshift/api#master)
 2. Rerun `hack/update-deps.sh` and update your `bump(*)` commit
 3. It can then be tagged and merged by CI
+
+## generating CRD schemas
+
+Since Kubernetes 1.16, every CRD created in `apiextensions.k8s.io/v1` is required to have a [structural OpenAPIV3 schema](https://kubernetes.io/blog/2019/06/20/crd-structural-schema/). The schemas provide server-side validation for fields, as well as providing the descriptions for `oc explain`. Structural requirements build on that to provide enhanced security in objects' persistence in etcd. As we host many of our CRDs in this repo along with their corresponding Go types we also require them to have schemas. However, the following instructions apply for CRDs that are not hosted here as well.
+
+These schemas are often very long and complex, and should not be written by hand. For OpenShift, we provide Makefile targets in [library-go's alpha-build-machinery](https://github.com/openshift/library-go/tree/master/alpha-build-machinery) which generate the schema.
```
Comment (sttts) [inline README.md]: would reference controller-tools as well as the tool behind all of this.

**Comment by damemi** [comment]
@sttts thanks for your review, I've added the changes you suggested

**Pull request merged**